src/coreclr/jit/gentree.cpp

```diff
@@ -14299,6 +14299,14 @@ GenTree* Compiler::gtFoldExprSpecial(GenTree* tree)
     // Here `op` is the non-constant operand, `cons` is the constant operand
     // and `val` is the constant value.
 
+    if (((op->gtFlags & GTF_SIDE_EFFECT) != 0) && ((tree->gtFlags & GTF_RELOP_JMP_USED) != 0) && tree->OperIsCompare())
```

Comment:
Super nit, just to avoid checking for a flag while it may overlap a flag for some other node type...
```suggestion
    if (((op->gtFlags & GTF_SIDE_EFFECT) != 0) && tree->OperIsCompare() && ((tree->gtFlags & GTF_RELOP_JMP_USED) != 0))
```

Reply:
I actually intentionally ordered it with `tree->OperIsCompare` last because the flag check is cheaper than the switch to check the oper kind, so it results in less code in the common case while still remaining correct.

Reply:
`OperIsCompare` is not a switch, it is a simple range check on the opers. I don't think we should be checking for oper-specific flags until we've verified those flags are valid for the oper we're dealing with. It introduces potential for mistakes during refactoring.

Reply:
Fixed.